src-test/org/etools/j1939_84/controllers/part1/Step08ControllerTest.java

```diff
@@ -0,0 +1,413 @@
+/**
+ * Copyright 2020 Equipment & Tool Institute
+ */
+package org.etools.j1939_84.controllers.part1;
+
+import static org.etools.j1939_84.model.Outcome.FAIL;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.stream.Collectors;
+
+import org.etools.j1939_84.bus.j1939.J1939;
+import org.etools.j1939_84.bus.j1939.packets.DM20MonitorPerformanceRatioPacket;
+import org.etools.j1939_84.bus.j1939.packets.PerformanceRatio;
+import org.etools.j1939_84.controllers.ResultsListener;
+import org.etools.j1939_84.controllers.TestResultsListener;
+import org.etools.j1939_84.model.FuelType;
+import org.etools.j1939_84.model.OBDModuleInformation;
+import org.etools.j1939_84.model.PartResultFactory;
+import org.etools.j1939_84.model.VehicleInformation;
+import org.etools.j1939_84.modules.BannerModule;
+import org.etools.j1939_84.modules.DateTimeModule;
+import org.etools.j1939_84.modules.DiagnosticReadinessModule;
+import org.etools.j1939_84.modules.EngineSpeedModule;
+import org.etools.j1939_84.modules.ReportFileModule;
+import org.etools.j1939_84.modules.TestDateTimeModule;
+import org.etools.j1939_84.modules.VehicleInformationModule;
+import org.etools.j1939_84.utils.AbstractControllerTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/**
+ *
+ * @author Garrison Garland (garrison@soliddesign.net)
+ *
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class Step08ControllerTest extends AbstractControllerTest {
```

Comment:
This needs a test to verify we behave correctly if no packets are returned from the vehicle.